### PR TITLE
Application runs through config.ru

### PR DESCRIPTION
**Pull request by gemcfadyen**
@jsuchy - The web application can be started using the command `rackup`. This will invoke the config.ru, which is the entry point for the web application.

tic_tac_toe_web_app.rb is now a class, rather than just the get methods. This class has been moved to within the lib folder, where all the classes live. This means that it is necessary to set the path to the views and public folders explicitly, otherwise by default, the class expects them relative to where it is (which would be lib/views, lib/public).

### CODE REVIEW

File: lib/tic_tac_toe_web_app.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  set :views, File.dirname(__FILE__) + '/../views'
+  set :public_folder, File.dirname(__FILE__) + '/../public'
+
+  get '/' do
+    @formatted_rows = OneIndexedGridFormatter::new.format(Board.new)
+    @valid_moves = PlayerSymbols::all
+    erb :landing_page
+  end
+
+  get '/next_move' do
```
Comment (felipesere) [inline lib/tic_tac_toe_web_app.rb:19]: I would split 'web stuff'  from 'ttt stuff'.
So I'd expect `get '/next_move` to to extract everything it needs from the request,
hand that data over and convert the result to something 'web-ish' like HTML or a redirect.

File: config.ru [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#Setup Load Path
+$LOAD_PATH << File.expand_path("../lib", __FILE__)
+
+require './lib/tic_tac_toe_web_app'
```
Comment (jsuchy) [inline config.ru]: Since you modified the load path right above, I think this should read `require 'tic_tac_toe_web_app'`. Avoid putting unneeded paths in your requires otherwise you won't be able to move files around.

File: lib/tic_tac_toe_web_app.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+require 'web_display_to_board_adapter'
+require 'web_game'
+
+class TicTacToeWebApp < Sinatra::Base
+
+  set :views, File.dirname(__FILE__) + '/../views'
+  set :public_folder, File.dirname(__FILE__) + '/../public'
+
+  get '/' do
+    @formatted_rows = OneIndexedGridFormatter::new.format(Board.new)
```
Comment (jsuchy) [inline lib/tic_tac_toe_web_app.rb]: ::new? why not .new?

File: lib/tic_tac_toe_web_app.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    @valid_moves = PlayerSymbols::all
+    erb :landing_page
+  end
+
+  get '/next_move' do
+    @move = params['move-taken']
+
+    contents_of_board = params['grid']
```
Comment (jsuchy) [inline lib/tic_tac_toe_web_app.rb:22]: There's a lot going on in here, and controllers are difficult to unit test. Consider extracting this into easier-to-test classes.

Comment (gemcfadyen) [inline lib/tic_tac_toe_web_app.rb:22]: Thanks Jim. Will do this update in a separate pull request. All the other comments I've addressed and added to this pr.
